examples/structured_2d_dgsem/elixir_mhd_coupled.jl

```diff
@@ -0,0 +1,138 @@
+using OrdinaryDiffEq
+using Trixi
+
+###############################################################################
+# Coupled semidiscretization of two ideal glmMhd systems using converter functions such that
+# they are also coupled across the domain boundaries to generate a periodic system.
+#
+# In this elixir, we have a square domain that is divided into a left and right half.
+# On each half of the domain, a completely independent SemidiscretizationHyperbolic is created for the
+# linear ideal glmMhd equations. The four systems are coupled in the x and y-direction.
+# For a high-level overview, see also the figure below:
+#
+# (-2,  2)                                   ( 2,  2)
+#     ┌────────────────────┬────────────────────┐
+#     │    ↑ periodic ↑    │    ↑ periodic ↑    │
+#     │                    │                    │
+#     │     =========      │     =========      │
+#     │     system #1      │     system #2      │
+#     │     =========      │     =========      │
+#     │                    │                    │
+#     │<-- coupled         │<-- coupled         │
+#     │         coupled -->│         coupled -->│
+#     │                    │                    │
+#     │    ↓ periodic ↓    │    ↓ periodic ↓    │
+#     └────────────────────┴────────────────────┘
+# (-2, -2)                                   ( 2, -2)
+
+
+equations = IdealGlmMhdEquations2D(1.4)
+
+cells_per_dimension = (32, 64)
+
+volume_flux = (flux_hindenlang_gassner, flux_nonconservative_powell)
+solver = DGSEM(polydeg = 3,
+               surface_flux = (flux_lax_friedrichs, flux_nonconservative_powell),
+               volume_integral = VolumeIntegralFluxDifferencing(volume_flux))
+
+function initial_condition_constant(x, t, equations::IdealGlmMhdEquations2D)
+  rho = 1.0
+  v1 = 0.0
+  v2 = 0.0
+  v3 = 0.0
+  p = rho^equations.gamma
+  B1 = 0.0
+  B2 = 0.0
+  B3 = 0.0
+  psi = 0.0
+  return prim2cons(SVector(rho, v1, v2, v3, p, B1, B2, B3, psi), equations)
+end
```

Comment:
Is this to test free-stream preservation? It is always dangerous to use ones and/or zeros to check free-stream because it can hide bugs. I would change these initial condition constants to be non-trivial values like `pi` or `exp(1)`.

Reply:
I will be using Alfven waves for testing. At least that is the plan. In my current implementation two systems can be coupled and Alfven waves travel from one system to the next. But there are a few thing I still need to iron out before settling on a test case.

Reply:
We are now using `initial_condition_convergence_test` as initial condition.